index.coffee

```diff
@@ -198,10 +198,10 @@ exports.DockerProgress = class DockerProgress
 				onProgress(evt)
 
 	# Pull docker image calling onProgress with extended progress info regularly
-	pull: (image, onProgress, callback) ->
+	pull: (image, onProgress, callback, options) ->
```

Comment:
It's a convention in Node for `callback` to be the last argument. That means `options` should come earlier, preferably as the second argument, after `image`. And because `options` is optional (and it'd break compatibility if it was required) we have to handle the different "virtual" method signatures by checking whether `options` is an object or a function and re-assign variables appropriately.